tests/vllm_test_utils/vllm_test_utils/monitor.py

```diff
@@ -0,0 +1,65 @@
+import contextlib
+import dataclasses
+import sys
+import traceback
+from typing import Any, Callable, Generator, List
+
+
+@dataclasses.dataclass
+class MonitoredValues:
+    values: List[Any] = dataclasses.field(default_factory=list)
+    trace_stacks: List[str] = dataclasses.field(default_factory=list)
+
+
+@contextlib.contextmanager
+def monitor(
+        measure_func: Callable[[],
+                               Any]) -> Generator[MonitoredValues, None, None]:
+    """
+    Trace the function calls to continuously monitor the change of
+    a value.
+
+    Usage:
+
+    ```python
+
+    def measure_func():
+        ... # measure the current value
+        return current_value
+
+    with monitor(measure_func) as monitored_values:
+        # do something
+    
+    monitored_values.values # all changes of the values
+    monitored_values.trace_stacks # trace stacks of every change
+    """
+    monitored_values = MonitoredValues()
```

Comment:
```suggestion
from typing import Any, Callable, Generator, Generic, TypeVar

_T = TypeVar("_T")

@dataclasses.dataclass
class MonitoredValues(Generic[_T]):
    values: list[_T] = dataclasses.field(default_factory=list)
    trace_stacks: list[str] = dataclasses.field(default_factory=list)


@contextlib.contextmanager
def monitor(
        measure_func: Callable[[],
                               _T]) -> Generator[MonitoredValues[_T], None, None]:
    """
    Trace the function calls to continuously monitor the change of
    a value.

    Usage:

    ```python

    def measure_func():
        ... # measure the current value
        return current_value

    with monitor(measure_func) as monitored_values:
        # do something
    
    monitored_values.values # all changes of the values
    monitored_values.trace_stacks # trace stacks of every change
    """
    monitored_values = MonitoredValues[_T]()
```

We can keep the type information using this code.